### PR TITLE
IS-508: Introduce Interface for GrowthBook

### PR DESCRIPTION
**Pull request by harishv7**
## Problem

Previously, we used a constants file to keep track of the allowed feature flags. 

Closes IS-508

## Solution

This PR uses the recommended approach of using an interface instead. This still allows for type safety during development.

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [X] No - this PR is backwards compatible

### CODE REVIEW

File: src/types/featureFlags.ts [hunk 1]
```diff
@@ -1,15 +1,7 @@
...
-
-export type FeatureFlagTypes = typeof FeatureFlagSupportedTypes[keyof typeof FeatureFlagSupportedTypes]
-
-export type FeatureFlag = {
-  key: string
-  type: FeatureFlagTypes
+// Example usage: const gb = useGrowthBook<FeatureFlags>();
+export interface FeatureFlags {
+  sampleFlag: string
+  sampleFlag2: string
```
Comment (seaerchin) [inline src/types/featureFlags.ts:4]: are these supposed to be actual feature flags?

**Review by seaerchin — COMMENTED** [review]
not in favour of having example code in prod - would be ok with this PR if the sample stuff was commented otu

File: src/types/featureFlags.ts (hunk 1 above)
Comment (harishv7) [reply src/types/featureFlags.ts:4]: @seaerchin these are just samples. we will need to modify this when we actually create flags on growthbook

**Review by seaerchin — APPROVED** [review]
lgtm. please do comment out the samples

**Pull request merged**